### DISCUSSION TITLE
This provider is disabled.

### DISCUSSION THREAD

**Discussion by rafaelvicio** [nextauthjs/next-auth#1383] [Help]
**Describe the bug**
The error 'This server is disabled.' is being returned in the callback

**Steps to reproduce**
source: https://strapi.io/blog/user-authentication-in-next-js-with-strapi

```js
import NextAuth from "next-auth";
import Providers from "next-auth/providers";

const options = {
  providers: [
    Providers.GitHub({
      clientId: process.env.GITHUB_CLIENT_ID,
      clientSecret: process.env.GITHUB_CLIENT_SECRET,
    }),
  ],
  session: {
    jwt: true,
  },
  callbacks: {
    session: async (session, user) => {
      session.jwt = user.jwt;
      session.id = user.id;
      return Promise.resolve(session);
    },
    jwt: async (token, user, account) => {
      console.log("account", JSON.stringify(account));

      const isSignIn = user ? true : false;
      if (isSignIn) {
        const response = await fetch(
          `${process.env.NEXT_PUBLIC_API_URL}/auth/${account.provider}/callback?access_token=${account?.accessToken}`
        );
        const data = await response.json();

        console.log("data", JSON.stringify(data));
        token.jwt = data.jwt;
        token.id = data.user.id;
      }
      return Promise.resolve(token);
    },
  },
};

const Auth = (req, res) => NextAuth(req, res, options);

export default Auth;
```

**Expected behavior**

**Screenshots or error logs**
https://imgur.com/IYDaTeP

`account {"provider":"github","type":"oauth","id":10675033,"accessToken":"f99b96f25b7d08e2dbe5e4c1e28d8dd13a65578d","accessTokenExpires":null,"access_token":"f99b96f25b7d08e2dbe5e4c1e28d8dd13a65578d","scope":"user","token_type":"bearer"}
data {"statusCode":400,"error":"Bad Request","message":[{"messages":[{"id":"provider.disabled","message":"This provider is disabled."}]}],"data":[{"messages":[{"id":"provider.disabled","message":"This provider is disabled."}]}]}
`

**Additional context**
I tried again with Facebook Provider but the same problem was presented.

**Feedback**

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment]
The problem here comes from Strapi, as we don't emit any error message like `This provider is disabled`. What I think is that `token.id = data.user.id` fails, because your user object is undefined. If you console.log data in jwt, what do you get?. @ghoshnirmalya do you have any ideas what this could be?
